core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -788,6 +788,20 @@ abstract class RDD[T: ClassTag](
   }
 
   /**
+   * Applies a function to each partition of this RDD, while tracking the index
+   * of the original partition.
+   */
+  def foreachPartitionWithIndex(
```

Comment:
I disagree -- I think it should stay here.  It seems like a pretty obvious omission from the RDD api.  Probably its never been added b/c there wasn't a good use case.  Well, now we've got a good use case.  IMO no sense in creating another PR that needs review just for that.